test: Enable jest-console for native tests

dcalhoun opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->
Enable `@wordpress/jest-console` for the native editor automated tests.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Ensure accidental errors, warnings, and logs are not outputted during tests.
For example, an `act` warning will now cause a test failure.

Additionally, this will further align native editor tests with web editor tests. 

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
Enable the `@wordpress/jest-console` package within the native tests Jest
configuration, much like the web editor tests. Add assertions and to do comments
for all pre-existing logs that need to be addressed.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->
n/a

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->
n/a

## Screenshots or screencast <!-- if applicable -->
n/a

dcalhoun committed:
test: Enable jest-console for native tests

Ensure accidental errors, warnings, or logs are not outputted during
tests.

dcalhoun committed:
test: Assert Cover block accidental warning

dcalhoun committed:
test: Assert Embed block log

dcalhoun committed:
test: Assert Paragraph block accidental warning

dcalhoun committed:
test: Assert Buttons block accidental warning

dcalhoun committed:
test: Assert Heading block accidental warning

dcalhoun committed:
test: Assert text-color accidental warning

dcalhoun committed:
test: Address act warning in Editor test

Toggling HTML mode results in expected re-renders. We must wrap the
invocation in `act` to communicate that to Jest.

dcalhoun committed:
test: Replace unnecessary asynchronous query

Synchronous queries are generally easier to comprehend.

dcalhoun committed:
test: Assert block-invalid-warning warning and error

dcalhoun committed:
test: Assert Draggable accidental warning

dcalhoun committed:
test: Assert setup-local logs

dcalhoun commented on packages/edit-post/src/test/editor.native.js:
```diff
@@ -74,11 +74,12 @@ describe( 'Editor', () => {
 		fireEvent.press( paragraphBlock );
-
-		toggleMode();
+		act( () => {
+			toggleMode();
+		} );
```
Toggling HTML mode results in intentional updates and re-renders. We must wrap this call in `act` to communicate that intent to Jest.

dcalhoun commented on packages/react-native-editor/src/test/setup-locale.test.js:
```diff
@@ -28,6 +28,11 @@ const pluginTranslations = [
+		expect( console ).toHaveLoggedWith( 'locale', 'test', {
+			...getDefaultTranslation(),
+			...extraTranslations,
+		} );
```
The related logs were added in https://github.com/wordpress-mobile/gutenberg-rnmobile-monorepo-commit-history/commit/b715c4de734d0497c5cca62e3c89c94b97130f8c. I would argue they are unnecessary noise and should be removed, or at least disabled behind a `DEBUG` or `VERBOSE` environment variable when running the development server. 

https://github.com/WordPress/gutenberg/blob/6e4bc840699ed2742dba27e585b38ee136f882be/packages/react-native-editor/src/setup-locale.js#L38-L46

fluiddot replied:
> I would argue they are unnecessary noise and should be removed, or at least disabled behind a `DEBUG` or `VERBOSE` environment variable when running the development server.

I agree it's quite noisy, especially when debugging and checking the log output. I understand this log was added to check if the correct translations are being passed when opening the editor, however, in my case I've never used it. 
Based on this issue https://github.com/wordpress-mobile/gutenberg-mobile/issues/1822, seems we already considered removing this log, so I'd advocate finally removing it.

dcalhoun replied:
Addressed in 91bff10f4e816acd52c34dc8f1ad1a8a2ae9775c.

fluiddot commented on test/native/setup-after-env.js:
```diff
@@ -1,3 +1,8 @@
+/**
+ * WordPress dependencies
+ */
+import '@wordpress/jest-console';
```
Now that we import this module for all tests, we can consider removing the import from test cases where it was added:

https://github.com/WordPress/gutenberg/blob/c8b392a8404889b296e02ed9e8f30ae77a3bd6f4/packages/block-editor/src/components/block-draggable/test/index.native.js#L19
https://github.com/WordPress/gutenberg/blob/c8b392a8404889b296e02ed9e8f30ae77a3bd6f4/packages/block-library/src/image/test/edit.native.js#L29
https://github.com/WordPress/gutenberg/blob/c8b392a8404889b296e02ed9e8f30ae77a3bd6f4/packages/react-native-editor/src/test/index.test.js#L14
https://github.com/WordPress/gutenberg/blob/c8b392a8404889b296e02ed9e8f30ae77a3bd6f4/packages/rich-text/src/test/index.native.js#L18

dcalhoun replied:
Addressed in c28087f5afd8530076aa8ffa16813b9cd3706d63.

fluiddot reviewed (approved):
LGTM 🎊 !

I've added a comment regarding removing no-longer-needed imports, but shouldn't block merging this PR.

dcalhoun committed:
test: Remove redundant jest-console imports

This library is now included universally for all tests via test
environment configuration and setup files.

dcalhoun committed:
refactor: Remove translation log

The log introduces additional, un-filterable noise that can impair
one's ability to use logs for debugging.

fluiddot reviewed (approved):
I've reviewed the last changes and look good to me :shipit: .
